embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/testcases/SessionApiTest.kt

```diff
@@ -58,41 +57,37 @@ internal class SessionApiTest {
                 // validate network status span
                 val networkStatusSpan = snapshots.single { it.name == "emb-network-status" }
                 assertEquals(startTime, networkStatusSpan.startTimeNanos?.nanosToMillis())
-                assertEquals("sys.network_status", networkStatusSpan.attributes?.findAttributeValue("emb.type"))
+                networkStatusSpan.attributes?.assertMatches {
+                    "emb.type" to "sys.network_status"
+                }
 
                 // validate session span
                 val spans = checkNotNull(message.data.spans)
                 val sessionSpan = spans.single { it.name == "emb-session" }
                 assertEquals(startTime, sessionSpan.startTimeNanos?.nanosToMillis())
                 assertNotNull(sessionSpan.attributes?.findAttributeValue(SessionIncubatingAttributes.SESSION_ID.key))
+
+                sessionSpan.attributes?.assertMatches {
+                    "emb.cold_start" to true
+                    "emb.state" to "foreground"
+                    "emb.clean_exit" to true
+                    "emb.session_start_type" to "state"
+                    "emb.terminated" to false
+                    "emb.session_end_type" to "state"
+                    "emb.heartbeat_time_unix_nano" to startTime.millisToNanos()
+                    "emb.session_number" to 1
+                    "emb.type" to "ux.session"
+                    "emb.error_log_count" to 0
+                    "emb.usage.set_username" to 1
+                    "emb.usage.set_user_email" to 1
+                    "emb.usage.set_user_identifier" to 1
+                    "emb.private.sequence_id" to 4
+                }
                 val attrs = checkNotNull(sessionSpan.attributes)
                 val attributeKeys = attrs.map { it.key }
                 validateExistenceOnly.forEach { key ->
                     attributeKeys.contains(key)
                 }
-
-                val attributesToCheck = attrs.filterNot {
-                    ignoredAttributes.contains(it.key)
-                }.toMap().toSortedMap()
-
-                val expected = mapOf(
```

Comment:
The original idea here is to ensure that no new attributes show up unexpectedly, so we explicitly ignore ones that may or may not be there because of race conditions. While what makes it more of a "gold file" diff, it also can get flakey. Do we want to preserve this or nah?

Reply:
Hmm - I can see this still being useful. Perhaps we can add in an extra assertion after `assertMatches` that explicitly finds any new/missing attributes?

Reply:
What are you thinking in terms of syntax? Something like this?

```
fun assertMatches(allSpecified: Boolean, expected: MutableMap<String, Any>.() -> Unit) {
...
}
```

Reply:
I figured it's simplest just to leave it as-is for now: https://github.com/embrace-io/embrace-android-sdk/pull/1461